src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ValidateVariants.java

```diff
@@ -242,19 +261,29 @@ public void apply(final VariantContext vc, final ReadsContext readsContext, fina
 
             validateVariantsOrder(vc);
 
+            final boolean thisIntervalIsReference =  vc.getAlternateAlleles().size() == 1 && vc.getAlternateAllele(0).equals(Allele.NON_REF_ALLELE);
+
             // GenomeLocSortedSet will automatically merge intervals that are overlapping when setting `mergeIfIntervalOverlaps`
             // to true.  In a GVCF most blocks are adjacent to each other so they wouldn't normally get merged.  We check
             // if the current record is adjacent to the previous record and "overlap" them if they are so our set is as
             // small as possible while still containing the same bases.
+            if (previousInterval != null && previousInterval.overlapsWithMargin(refInterval, 0) &&
+                    (previousIntervalIsReference || thisIntervalIsReference)) {
+                logger.warn("Current interval " + refInterval.toString() + " overlaps previous interval " + previousInterval.toString());
+                sawOverlap = true;
+            }
             final int start = (previousInterval != null && previousInterval.overlapsWithMargin(refInterval, 1)) ?
                     previousInterval.getStart() : refInterval.getStart();
             final int end = (previousInterval != null && previousInterval.overlapsWithMargin(refInterval, 1)) ?
                     Math.max(previousInterval.getEnd(), vc.getEnd()) : vc.getEnd();
             final GenomeLoc possiblyMergedGenomeLoc = genomeLocSortedSet.getGenomeLocParser().createGenomeLoc(refInterval.getContig(), start, end);
             genomeLocSortedSet.add(possiblyMergedGenomeLoc, true);
 
-            previousInterval = new SimpleInterval(possiblyMergedGenomeLoc);
-            previousStart = vc.getStart();
+            //if (vc.getGenotype(0).isHomRef()) {
+                previousInterval = new SimpleInterval(possiblyMergedGenomeLoc);
```

Comment:
I think you may actually need to keep a separate track of ref blocks here. If you imagine a bam with the intervals 100-205, a variant at 201, then the intervals 202-300 that would be an illegal ref block overlap but since 201 is not reference `thisIntervalIsReference = false` so it doesn't trigger an overlap, then we save the previuous 201 interval in previous `previousInterval = new SimpleInterval(possiblyMergedGenomeLoc);` and then when we hit the ref block 202-300 we will again fail to detect the overlap here because we are comparing ourselves to the last variant site and not the previously illegal overlapping ref block.

Reply:
If 100-205 is a ref block and 201 is a variant, then the 100-205 ref block is bad.